stdlib/test/memory/test_boxed.mojo

```diff
@@ -0,0 +1,129 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+# RUN: %mojo --debug-level full %s
```

Comment:
```suggestion
# RUN: %mojo  %s
```

Reply:
For types that may have debug asserts within them, do we not want to test with those all enabled?

Reply:
As surprising as it may sound, the `--debug-level full` is rather around creating the debug info in the IR.  We have some additional testing for this internally that parameterizes how `%mojo` expands, similar to how we do this in the open source CI for running with/without assertions.  See https://github.com/modularml/mojo/blob/936018ed2980488cd0e10da9298279e01c563137/.github/workflows/standard_library_tests_and_examples.yml#L31 for example - it gets propagated at https://github.com/modularml/mojo/blob/936018ed2980488cd0e10da9298279e01c563137/stdlib/test/lit.cfg.py#L73.